src/Npgsql/TypeHandlers/CompositeHandlers/MappedCompositeHandler.cs

```diff
@@ -75,14 +90,16 @@ public override int ValidateAndGetLength(T value, ref NpgsqlLengthCache? lengthC
             lengthCache.Set(0);
 
             // number of fields + (type oid + field length) * member count
-            var length = sizeof(int) + sizeof(int) * 2 * _members.Length;
-            foreach (var member in _members)
+            var length = sizeof(int) + sizeof(int) * 2 * _memberHandlers.Length;
+            foreach (var member in _memberHandlers)
                 length += member.ValidateAndGetLength(value, ref lengthCache);
 
             return lengthCache.Lengths[position] = length;
         }
 
-        public static MappedCompositeHandler<T> Create(PostgresCompositeType pgType, ConnectorTypeMapper typeMapper, INpgsqlNameTranslator nameTranslator)
+        CompositeMemberHandler<T>[] CreateMemberHandlers() => CreateMemberHandlers((PostgresCompositeType)PostgresType, _typeMapper, _nameTranslator);
```

Comment:
nit: I don't think there's any reason to have the static `CreateMemberHandlers()` below - it's only being called here. Maybe just merge the code into this method?

Reply:
I left the static method to have immutable version of it, and the instance one is for simplification only. What do you think about it?

Reply:
What do you mean by immutable? Can you provide more detail?

(I'm also OK with merging as-is if you don't want to go into it, it's really not so important)

Reply:
It's a method which doesn't modify an object state, i.e. a `const` function in C++.

Reply:
This could be changed too, because there will be two methods to initialize a handler. Merging as is.

Reply:
OK. I don't really understand the logic but it's not important.